bot/exts/info/tags.py

```diff
@@ -172,125 +246,163 @@ async def search_tag_content(self, ctx: Context, *, keywords: str) -> None:
+            if tag.accessible_by(ctx.author)
+        ]
+
+        if (tag := self._tags.get(tag_identifier)) is None:
+            if len(filtered_tags) == 1:
+                tag_identifier = filtered_tags[0][0]
+                tag = filtered_tags[0][1]
+
+        if tag is not None:
+            if tag.on_cooldown_in(ctx.channel):
+                log.debug(f"Tag {str(tag_identifier)!r} is on cooldown.")
+                return COOLDOWN.obj
+            tag.set_cooldown_for(ctx.channel)
+
+            self.bot.stats.incr(
+                f"tags.usages"
+                f"{'.' + tag_identifier.group.replace('-', '_') if tag_identifier.group else ''}"
+                f".{tag_identifier.name.replace('-', '_')}"
             )
+            return tag.embed
 
-            if cooldown_conditions:
-                return True
-            return False
-
-        if _command_on_cooldown(tag_name):
-            time_elapsed = time.time() - self.tag_cooldowns[tag_name]["time"]
-            time_left = constants.Cooldowns.tags - time_elapsed
-            log.info(
-                f"{ctx.author} tried to get the '{tag_name}' tag, but the tag is on cooldown. "
-                f"Cooldown ends in {time_left:.1f} seconds."
+        else:
+            if not filtered_tags:
+                return None
+            suggested_tags_text = "\n".join(
+                f"**\N{RIGHT-POINTING DOUBLE ANGLE QUOTATION MARK}** {identifier}"
+                for identifier, tag in filtered_tags
+                if not tag.on_cooldown_in(ctx.channel)
+            )
+            return Embed(
+                title="Did you mean ...",
+                description=suggested_tags_text
             )
-            return True
-
-        if tag_name is not None:
-            temp_founds = self._get_tag(tag_name)
-
-            founds = []
-
-            for found_tag in temp_founds:
-                if self.check_accessibility(ctx.author, found_tag):
-                    founds.append(found_tag)
-
-            if len(founds) == 1:
-                tag = founds[0]
-                if ctx.channel.id not in TEST_CHANNELS:
-                    self.tag_cooldowns[tag_name] = {
-                        "time": time.time(),
-                        "channel": ctx.channel.id
-                    }
 
-                self.bot.stats.incr(f"tags.usages.{tag['title'].replace('-', '_')}")
+    async def list_all_tags(self, ctx: Context) -> None:
+        """Send a paginator with all loaded tags accessible by `ctx.author`, groups first, and alphabetically sorted."""
+        def tag_sort_key(tag_item: tuple[TagIdentifier, tag]) -> str:
+            ident = tag_item[0]
+            if ident.group is None:
+                # Max codepoint character to force tags without a group to the end
+                group = chr(0x10ffff)
+            else:
+                group = ident.group
+            return group+ident.name
+
+        result_lines = []
+        current_group = object()
```

Comment:
Is this also a sentinel value? Can you not use a empty string (I know that None is used when a tag doesn't have a group).

Reply:
I suppose an empty string could be used, but I don't think there's any benefit of using it instead of a plain object like the current use

Reply:
Well there's a benefit in annotating the code